File: twosum.py
class Solution:
    def twoSum(self, nums, target):
        theOtherDict = {}
        for index, item in enumerate(nums):
            if item in theOtherDict:
                return [theOtherDict[item], index]
            else:
                theOtherDict[target - item] = index


def execute():
    nums = [2, 7, 11, 15]
    target = 9
    sol = Solution()
    print(sol.twoSum(nums=nums, target=target))


if __name__ == "__main__":
    execute()

File: twosum.js

function twoSum(nums, target) {
  let theOtherDict = {};
  let theResult = {};
  nums.forEach((value, index, array)=> {
      if (value in theOtherDict)
      {
        theResult[theOtherDict[value]] = index;
      } else {
        theOtherDict[target - value] = index;
      }
    });
  return theResult;
}

function execute() {
  let nums = [2, 7, 11, 15];
  let target = 9;
  console.log(twoSum(nums, target));
}

execute();
